apps/prairielearn/src/lib/assessment.ts

```diff
@@ -365,11 +373,23 @@ export async function selectAssessmentInstanceLog(
   assessment_instance_id: string,
   include_files: boolean,
 ): Promise<InstanceLogEntry[]> {
-  return sqldb.queryRows(
+  const log = await sqldb.queryRows(
     sql.assessment_instance_log,
     { assessment_instance_id, include_files },
     InstanceLogSchema,
   );
+  const fingerprintNumbers = {};
+  let i = 1;
+  log.forEach((row) => {
+    if (row.client_fingerprint) {
+      if (!fingerprintNumbers[row.client_fingerprint.id]) {
+        fingerprintNumbers[row.client_fingerprint.id] = i;
+        i++;
+      }
+      row.client_fingerprint_number = fingerprintNumbers[row.client_fingerprint.id];
+    }
+  });
+  return log;
 }
 
 export async function selectAssessmentInstanceLogCursor(
```

Comment:
This is the function that's used to produce the CSV; we use a streaming query to minimize memory. The fix here is unfortunately non-trivial, you'd need a streaming transformer for each object emitted here. Given that we want to ship this ASAP, I'm inclined to _not_ include the fingerprint number in the CSV for now and create a followup ticket to track that work instead.

Reply:
I added this to https://github.com/PrairieLearn/PrairieLearn/issues/8924

Reply:
I have removed `client_fingerprint` from the CSV in commit [d9968ea](https://github.com/PrairieLearn/PrairieLearn/pull/8798/commits/d9968ea4a3acf77b35cb47c0c08bb0cdf36f7246) and I will add this feature when I am working on #8924 .